docs/source/guide/Modules.lrad

```diff
@@ -0,0 +1,107 @@
+# Modules
+
+To create a module, use the `module` special form. All modules must start with a
+*module declaration* which provides the module with a name, documentation string
+and an exports vector. Module declarations are evaluated, so you must quote any
+
+Here is a simple example of a module:
+
+    (module
+      {:module  'my-module
+       :doc     "Just for showing off modules."
+       :exports '[foo bar]}
+
+      (def foo-helper
+        (fn [x] (+ x 1)))
+
+      (def foo
+        (fn [x] (foo-helper (foo-helper x))))
+
+      (def bar
+        (fn [x] (foo (+ x 2)))))
+
+To import a module, use `import`:
+
+    (import my-module)
+
+The definitions inside a module are evaluated in the current scope, so you can
+use whatever has already been defined. The form returns the module value, which
+can be passed around just like any other value. This means that we can write
+functions which create modules:
+
+    (def make-taxes
+      (fn [VAT]
+        (module
+         {:module 'taxes
+          :doc    "Calculates taxes."
+          :exports ['add-VAT]}
+         (def add-VAT (fn [x] (* (+ 1 VAT) x))))))
+
+And then we could use this module with different values of `VAT`:
+
+    (import (make-taxes 2/10))
+    ;; We also need to import `prelude/io` to be able to use `print!`
+    (import (file-module! "rad/prelude/io.rad") :unqualified)
+
+    (print! (string-append "100 EUR + VAT is: " (show (taxes/add-VAT 100))))
+
+By default `import` will add all the definitions of the module in
+fully-qualified form, which means `my-module/foo` and `my-module/bar` (but not
+`my-module/foo-helper`) are in scope:
+
+    (print! (my-module/bar 0))
+
+You can narrow down exactly which definitions you would like to import by
+including a list of atoms:
+
+    (import my-module ['foo])
+
+Which would only import `my-module/foo`. And you can also use a custom
+qualifier:
+
+    (import my-module 'baz)
```

Comment:
Would it be a lot of work to have clojure-style qualified imports (`(import my-module :as 'baz)`)? I imagine people expecting `(import my-module 'baz)` to import `baz` alone.